packages/medusa-payment-adyen/src/services/googlepay-adyen.js

```diff
@@ -26,12 +28,22 @@ class GooglePayAdyenService extends PaymentService {
     return status
   }
 
-  async createPayment(_) {
-    return {}
+  async createPayment(cart) {
+    const shippingOptions = await this.shippingProfileService_.fetchCartOptions(
+      cart
+    )
+    return {
+      shipping_options: shippingOptions,
```

Comment:
`fetchCartOptions` returns all the shipping options that are available regardless of shipping profile. When they are presented to the customer we should partition these by shipping profile so that the customer can choose 1 shipping option for regular items and one for other items e.g. gift cards. If the payment provider only allows the user to select one shipping option, we should calculate each of the combinations available. Take a look at https://github.com/medusajs/medusa/blob/d8483cd1352ecc587112723786b7c31882f9416e/packages/medusa-payment-klarna/src/services/klarna-provider.js#L162 to see how it is done in Klarna.

Reply:
I will actually delete this for now. We will investigate this again, when we need to make the change from directly using Klarna to using Klarna through Adyen.